src/issuance.rs

```diff
@@ -158,7 +158,7 @@ impl IssueAction {
 
     /// Serialize `finalize` flag to a byte
     pub fn flags(&self) -> u8 {
-        self.finalize.then(|| 0b0000_0001).unwrap_or(0b0000_0000)
+        u8::from(self.finalize)
```

Comment:
it will work, but the semantics are incorrect, we should specify the bit strings to clarify the intention.